Implementing DEEP Protocol marked as NotImplementedError

CelebrateVC opened the pull request:
I made this alteration to the library for my use back in May 2020 and thought I might as well share the change to expand this.

CelebrateVC committed:
refactor variable name to scope DEEP protocols too

Signed-off-by: Simon (Desktop) <celebrate.v.c@gmail.com>

CelebrateVC committed:
Parser can also take open file handle
useful for sublibary implementation (like I did)

Signed-off-by: Simon (Desktop) <celebrate.v.c@gmail.com>

CelebrateVC committed:
Implement new messages in DEEP 1.0

Signed-off-by: Simon (Desktop) <celebrate.v.c@gmail.com>

lvfrazao commented:
Hey @CelebrateVC thanks for contributing, due to my current employer's policies on open source code I need to get their approval before reviewing or merging this code.

snyk-bot committed:
fix: requirements.txt to reduce vulnerabilities


The following vulnerabilities are fixed by pinning transitive dependencies:
- https://snyk.io/vuln/SNYK-PYTHON-CERTIFI-3164749

CelebrateVC committed:
Merge pull request #1 from CelebrateVC/snyk-fix-88a00dc446f686bc444a8bd045f2a382

[Snyk] Security upgrade certifi from 2018.4.16 to 2022.12.7
